### DISCUSSION TITLE
Custom landing page to select Codespace template from multiple

### DISCUSSION THREAD

**Discussion by aaronsteers** [community/community#44949] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I'd like to create a custom version of something like the main template page at https://github.com/codespaces/templates, where multiple options are presented to the user via the same web URL:

> ![image](https://user-images.githubusercontent.com/18150651/213841839-f508ca6b-d339-4657-9b21-94f60a07b307.png)

-------------

The closest I can do as of now is using `template_repository` URL arg:

https://github.com/codespaces/new?template_repository=meltano/meltano-codespace-ready

> ![image](https://user-images.githubusercontent.com/18150651/213841878-a4e04fe5-928b-4c82-8727-e9c8f0276356.png)

Is it possible to make this pluralized, so that users can select from a few different options?

**Comment by joshspicer** [comment, marked as answer]
👋 Hey there.  We don't currently offer a way to contribute Codespace templates to that list on the `/codespaces` page.

There's a shortcut on template repos that you have probably seen to quickly create a codespace from a given template.
<img width="1773" alt="image" src="https://user-images.githubusercontent.com/23246594/214433501-53d22134-5217-4c9d-a30c-28415ee1022c.png">


----

I'd also suggest taking a look at [Dev Container Templates](https://containers.dev/templates) to see if that would be a possible solution for you.  Using this [starter repo](https://github.com/devcontainers/template-starter) you can self-author and publish templates that will show up on that list, and be visible via the Codespaces and Dev Containers VS Code extensions

<img width="735" alt="image" src="https://user-images.githubusercontent.com/23246594/214433979-9cc9256a-1604-4e4e-96fb-b3161d0c6855.png">

<img width="720" alt="image" src="https://user-images.githubusercontent.com/23246594/214434017-944bd512-caaa-4740-981b-4b98ee548be3.png">

We have many community contributions so far, and would additional template contributions if you think it would be a good fit!

**Reply by aaronsteers** [reply]
@joshspicer - Thanks very much for your reply. We are also working on the `Devcontainers Template` front for net-new projects using our platform. 

However, just to clarify my ask here, I would love to be able to provide a URL that shows multiple _tutorial repo_ options in the same UI.

It might not be something worthwhile adding the the central "Codespace Template" page, but it could be very nice for our new-user experience if we could give users a choice of "Start from scratch using our template" or "Start by exploring a pre-built project", or "Start with this advanced tutorial (experienced users)".

Something like

- `https://github.com/codespaces/new?template_choices=<myorg/repo1>,<myorg/repo2>,<myorg/repo3>`

And then ideally, these would render side-by-side as clickable options similar to the singular version we have today.

Does that make sense? And, understanding it might not exist now, is this something that might be added in future?

**Reply by joshspicer** [reply]
Gotcha, glad to hear you're playing with dev container templates :)

I'll forward the feedback/ask for a "templates landing page" to our product team 👍

**Reply by aaronsteers** [reply]
Fantastic - thanks, @joshspicer!

I'll wrap by saying we _love_ Codespaces and Codespaces templates. ❤️  We're very huge fans of this feature here at Meltano.

**Comment by aaronsteers** [comment]
Hi, @joshspicer - It looks like our old syntax for creating a new Codespace from a repo template has broken.

We were using this syntax:

https://github.com/codespaces/new?template_repository=meltano/meltano-codespace-ready

The above doesn't seem to understand templates anymore and instead will prompt the user to select a repo to use for the new codespace. Which also is the same URL posted to the top of this post. (From what I can tell, this clones the existing repo rather than starting a new repo from the template.)

Is there a new syntax we should use?

**Reply by aaronsteers** [reply]
Logged as new discussion for the bug report:

- [Deeplink syntax appears broken for New Codespace from Template · community · Discussion #53816](https://github.com/orgs/community/discussions/53816)

**Reply by joshspicer** [reply]
Thanks for bringing this up.  Brian responded over on your new discussion.
